### ISSUE TITLE
maybe a live CD would be helpful?

### ISSUE THREAD

**Issue by alexmyczko** [trunkmaster/nextspace#242]
would that be hard to create? i have no idea of non-debian based linux distributions.

**Comment by trunkmaster** [comment]
I've never created a live CD. Do you know how to create Debian based live CD?

It would be great if you create live CD and share results of you work (build scripts, configuration files)! Could you?

**Comment by alexmyczko** [comment]
Yeah I do that http://livecd.gnustep.org after all, but it was painful everytime different, and I'm behind 2 years.

For Fedora/CentOS you might want to check: https://wiki.centos.org/HowTos/InstallFromUSBkey

**Comment by nicktelindert** [comment]
I have some spare time so i could look into it

**Comment by trunkmaster** [comment]
@alexmyczko, let's change title to "Create Live CD". So this topic can be used as task and discussion.
I assign this issue to @nicktelindert.

**Comment by nicktelindert** [comment]
I will start with it tonight, i think i have a basic idea about how to do it.
- I first create one for boot only, so without any installation options
- Later goal is to look into installation posiblities

**Comment by trunkmaster** [comment]
@nicktelindert I guess boot only will be great step ahead. Do not hesitate to ask questions if you face with problems.

**Comment by nicktelindert** [comment]
@trunkmaster I think i use the livecd-tools from redhat, which seem pretty easy to use. But i need DNF for that. I probably use a virtualmachine to build the ISO because i run Debian 10 on  my system. I hope to have a test ISO in the upcoming week

**Comment by nicktelindert** [comment]
@trunkmaster BTW building NextSpace  in CentOS 8 or RHEL seems nearly impossible. They removed alot of development packages which are required to build NextSpace. And for most of them no third party repository was available. I mostly used pkgs.org to search for third party packages.

**Comment by trunkmaster** [comment]
@nicktelindert I know that RedHat removed some packages from RHEL/CentOS 8. It's better to build live CD with CentOS 7. I've made missed packages for CentOS 8 by myself using srpms from 7 (it's not that hard). I'll try to find specs and send you.

**Comment by nicktelindert** [comment]
I am using CentOS 7 now in a VM, the kernel is too old to install on my xps 13 that's why i've tried CentOS 8. I hadn't thought of using the srpms of centos 7. Thanks for mentioning it.

**Comment by nicktelindert** [comment]
I put my Kickstart script on my own github: https://github.com/nicktelindert/nextspace-build-iso

Not tested yet, am still working on it.

**Comment by nicktelindert** [comment]
A test ISO: https://drive.google.com/open?id=1s-jG79uuR0hyrexpZFo7YE1LQNNuLxGw
Please help me test and report issues.

**Comment by BertrandDekoninck** [comment]
Great ! Tested in Virtualbox. I can set the monitor resolution, the keyboard layout, choose the locale (with no effect as Workspace isn’t translated yet in French). I didn't manage to have sound yet, and the iso misses a bunch of apps (and app wrappers too) to demonstrate the system could be usefull. But Nextspace itself is impressive.

Bertrand Dekoninck

**Comment by nicktelindert** [comment]
> Great ! Tested in Virtualbox. I can set the monitor resolution, the keyboard layout, choose the locale (with no effect as Workspace isn’t translated yet in French). I didn't manage to have sound yet, and the iso misses a bunch of apps (and app wrappers too) to demonstrate the system could be usefull. But Nextspace itself is impressive.
> 
> Bertrand Dekoninck

- I dont think the workspace has any translations yet.

- for the base apps i have an issue open. My first goal was getting the basics work. Do you have any suggestions for some base apps? I need to fix some issues with my app wrapper too. It seems to have some issues with centos 7. 

I look into the audio issue, do you know which chip you used in virtualbox?

**Comment by nicktelindert** [comment]
For suggestions on base apps i recommend to post suggestions here:
https://github.com/nicktelindert/nextspace-build-iso/issues/3

**Comment by HankG** [comment]
I didn't see where the default login was listed anywhere (it didn't auto-login for me). I found it in the build script:
username=`nextspace`
password= ``

**Comment by nicktelindert** [comment]
Its listed on my github page

Op za 29 feb. 2020 23:08 schreef Hank G <notifications@github.com>:

**Comment by trunkmaster** [comment]
> A test ISO: https://drive.google.com/open?id=1s-jG79uuR0hyrexpZFo7YE1LQNNuLxGw
> Please help me test and report issues.

Execlent! I've tested in VirtualBox - everything works.  Well done.
Although after login I've got alerts from Workspace about inability to mount something? Is it some specific for livecd?

Some ideas:
- It will be good to have nextspace plymouth theme visible in VirtualBox. For this we need to have `vboxvideo` drive included into `initrd`.
- I see your build script installs Inkscape - would be great to have an app wrapper in /Applications.
- SELinux is not disabled in script - may result some problems. I guess it's better to disable it as described in "Install Guide" (wiki).

**Comment by HankG** [comment]
I can't seem to bring up the preferences app to set the display size. I installed it to a HD to make sure that wasn't what was causing the problem. I got the same behavior. It looks like it launches but I never get the actual window.

**Comment by trunkmaster** [comment]
@HankG, something wrong with UI model files (gorm). Look at Console for error messages (Workspace->Tools->Console).

**Comment by HankG** [comment]
No not seeing any. The launching is after all the "Dock iconMouseDown"
![image](https://user-images.githubusercontent.com/1683691/75616001-26b61a80-5b19-11ea-981f-cab93425558a.png)

**Comment by trunkmaster** [comment]
Have you double-clicked on Preferences application icon in Dock?

**Comment by HankG** [comment]
> Have you double-clicked on Preferences application icon in Dock?

Doh...no I've been going through applications each time...

**Comment by trunkmaster** [comment]
Preferences panel opens only after double-click on appicon - it's a designed behaviour.

**Comment by HankG** [comment]
I see that now :). OPENSTEP it did both ways but current behavior makes sense.

**Comment by trunkmaster** [comment]
You're right. Probably I need to make it behave as OPENSTEP did.

**Comment by HankG** [comment]
> You're right. Probably I need to make it behave as OPENSTEP did.

Not necessarily but for this it was a bit confusing. PS about to post this to my social media feeds and link to your project: 
![Screenshot from 2020-02-29 17-50-26](https://user-images.githubusercontent.com/1683691/75616183-28351200-5b1c-11ea-8b24-e9ce8fac4ba8.png)

**Comment by trunkmaster** [comment]
Cool. BTW, you can change title of Terminal window to match the OPENSTEP screenshot. :-)

**Comment by nicktelindert** [comment]
> > A test ISO: https://drive.google.com/open?id=1s-jG79uuR0hyrexpZFo7YE1LQNNuLxGw
> > Please help me test and report issues.
> 
> Execlent! I've tested in VirtualBox - everything works. Well done.
> Although after login I've got alerts from Workspace about inability to mount something? Is it some specific for livecd?
> 
> Some ideas:
> 
> * It will be good to have nextspace plymouth theme visible in VirtualBox. For this we need to have `vboxvideo` driver included into `initrd`.
> * I see your build script installs Inkscape - would be great to have an app wrapper in /Applications.
> * SELinux is not disabled in script - may result some problems. I guess it's better to disable it as described in "Install Guide" (wiki).

There is some stuff to be done indeed. On my own version i have included my app wrapper script and some basic apps. Selinux should be disabled though. I added selinux - - disabled in the build script. Main priority now is fixing some bugs and than fill some missing gaps by adding easier wifi control with nm-applet and wmsystemtray. I hope to add some more gnustep apps to create a more complete experience.

**Comment by trunkmaster** [comment]
> 
> Main priority now is fixing some bugs and than fill some missing gaps by adding easier wifi control with nm-applet and wmsystemtray. 

Sure, it's an essential parts to make LiveCD usable after install. However I hope Network Preferences module will be implemented anytime soon. It's my next big thing in NEXTSPACE development. Currently I'm busy with finalizing multi-monitor support for GNUstep and adopt it in NEXTSPACE.

> I hope to add some more gnustep apps to create a more complete experience.

I've sent you my set of app wrappers in https://github.com/nicktelindert/nextspace-build-iso/issues/3 discussion.

**Comment by doublet30** [comment]
I just tested the ISO in virtualbox, it was impressive in my honest opinion.  I was wondering if you had plans for a CI / CD system.  For example, this project could be configured to create packages for all distribution types.  RPM for RPM based distros, Deb for Debian Based distros, and Aur for Arch.  Package maintainers in those distros could then create CI / CD scripts that look for these packages, and include them into contrib repos when a "stable" release is published, or to an unstable repo when packages are automatically produced when changes are made.  Continuous Integration and Deployment can drastically help push a project like this forward and make a presence in all major distributions and derivatives.  A bigger presence might also mean more developer contributions.  To be honest, I stumbled on this project by just browsing github out of boredom.  I'm a software developer for Linux myself and a distro maintainer.  I have a soft spot for nostalgic DE's and this just happen to catch my eye.  Unfortunately, I have too much of a work load to contribute to this project otherwise it would be the first band wagon I would choose to hop on.  But a bigger presence might attract more help.  I would love to see this project completed and gaining popularity.  Especially since there has been no new work or innovation on Etoile OS in 6 years.  Being a smaller dev team, you made a more "functional" WindowMaker / GNUstep DE fork than the guys at EtoileOS did.  Kudo's for all the hard work guys!  It is definitely exciting and incredible! :)

**Comment by trunkmaster** [comment]
@doublet30, I'm glad you found my project interesting. It wil be great to have CI/CD and automatic package building. Unfortunately, I'm busy writing code at this stage of project. Many thanks to @nicktelindert for building the ISO image so more people can try NEXTSPACE. Without @nicktelindert help no LiveCD image will exist.

If you know how to setup CI/CD and spare tiny portion of your time - we'll get huge results. Current project state is a result of one developer - me. I do it on my spare time but everyday. I hope you're able to find some time to help us bring the NEXTSPACE to world. It's funny anyway.

**Comment by nicktelindert** [comment]
@trunkmaster I do it with pleasure:) I did setup a jenkins server for CI never used it for CD though. But i leave that task to someone else, but i could assist someone if they want to do the task with Jenkins.

**Comment by nicktelindert** [comment]
I do have some issues with the plymouth splash. I changed it in my build script and than run dracut - f but somehow it doesnt change the theme. Does anybody have a suggestion?

**Comment by trunkmaster** [comment]
> @trunkmaster I do it with pleasure:) I did setup a jenkins server for CI never used it for CD though. But i leave that task to someone else, but i could assist someone if they want to do the task with Jenkins.

If it is configured like Travis you can add config file via pull request.

**Comment by trunkmaster** [comment]
> I do have some issues with the plymouth splash. I changed it in my build script and than run dracut - f but somehow it doesnt change the theme. Does anybody have a suggestion?

For VM you need to add kernel module with supported KMS video driver (vboxvideo). For VMware kernel already conains appropriate video driver.

**Comment by nicktelindert** [comment]
> > I do have some issues with the plymouth splash. I changed it in my build script and than run dracut - f but somehow it doesnt change the theme. Does anybody have a suggestion?
> 
> For VM you need to add kernel module with supported KMS video driver (vboxvideo). For VMware kernel already conains appropriate video driver.

Those are already installed in my build, and the plymouth splash works fine now after installation but not on the livecd itself. So i will look into it later, now i try to figure out how to add the user created by anaconda to the audio group automatically.

**Comment by BertrandDekoninck** [comment]
This live-cd is a great way to test NextSpace. I've installed the system in VirtualBox using the anaconda installer provided by the live-cd. It works like a charm, much more efficient than a classical Centos install followed by the install of nextspace packages.
But there are some rough edges.

• I miss the french mac keyboard in NextSpace Preferences app : it isn't listed, even if I selected it in Anaconda. Should I file an issue against Preferences ?
• I've reported an issue with new users home creation here [https://github.com/nicktelindert/nextspace-build-iso/issues/14]. The folder is created in /home instead of /Users.
• I wanted to compile  rik theme but gnustep.sh isn't sourced and I could not find where it is located in NextSpace filesystem. This should be automatically added to new users environment variables at the creation of the users. Isn't this the in the home's default skeleton  ?

Last two issues which are not related to CD itself, but I report them first here.
• I could test more extensively the drag and drop with Workspace, but did not manage at first glance to move or copy a file/directory between Workspace's windows. I eventually manage to achieve move by dragging an icon from the shelf of the first window to the shelf of the second window.  A drag between the windows contents themselves doesn't work. 
I don't know how to copy either. The copy menuentry doesn't work . 
Is this the intended behaviour or something to adressed ? I don't think it is very usable for now and it's difficult to figure out. I only tried the shelf because I know it's it's a way to do in GWorkspace, but other people will be disappointed.

• I wanted to define Workspace as default program for Firefox to open folders (in the download manager), but I could'nt find the correct way in Firefox dialog : I selected Workspace executable in Workspace.app folder but this doesn't work. Should I file an issue elsewhere ?
I've got other issues with Workspace and themes I could discuss later.

**Comment by trunkmaster** [comment]
> This live-cd is a great way to test NextSpace. I've installed the system in VirtualBox using the anaconda installer provided by the live-cd. It works like a charm, much more efficient than a classical Centos install followed by the install of nextspace packages.
> But there are some rough edges.
> 
> • I miss the french mac keyboard in NextSpace Preferences app : it isn't listed, even if I selected it in Anaconda. Should I file an issue against Preferences ?

Do you mean Apple keyboard with French layout? Then you should make setup in Keyboard Preferences: "Model" section for keyboard type and "Layouts" section for French layout.

> • I've reported an issue with new users home creation here [https://github.com/[nicktelindert/nextspace-build-iso/issues/14](https://github.com/nicktelindert/nextspace-build-iso/issues/14)]. The folder is created in /home instead of /Users.
> • I wanted to compile rik theme but gnustep.sh isn't sourced and I could not find where it is located in NextSpace filesystem. Isn't gnustep-make installed ? Should I install libobjc and clang as well ?

Obviously development packages is not installed. nextspace*-dev RPMS place all necessary files into /Developer directory. Environment will be set up with /etc/profile.d/nextspace.sh file automatically at shell launch (for example in Terminal).
Please create issue at @nicktelindert project page.

> 
> Last two issues which are not related to CD itself, but I report them first here.
> • I could test more extensively the drag and drop with Workspace, but did not manage at first glance to move or copy a file/directory between Workspace's windows. I eventually manage to achieve move by dragging an icon from the shelf of the first window to the shelf of the second window. A drag between the windows contents themselves doesn't work.
> I don't know how to copy either. The copy menuentry doesn't work .
> Is this the intended behaviour or something to adressed ? I don't think it is very usable for now and it's difficult to figure out. I only tried the shelf because I know it's it's a way to do in GWorkspace, but other people will be disappointed.

Please read Workspace's help (Cmd-?) "Working with Files and Folders" section.

> 
> • I wanted to define Workspace as default program for Firefox to open folders (in the download manager), but I could'nt find the correct way in Firefox dialog : I selected Workspace executable in Workspace.app folder but this doesn't work. Should I file an issue elsewhere ?

It's not supported yet. Please create and issue in this project.

> I've got other issues with Workspace and themes I could discuss later.

**Comment by nicktelindert** [comment]
@trunkmaster i first finish some issues, development packages will be added in the next release. I find it difficult to say when i will do a next test release. I first want to fix stuff that i consider showstoppers.

**Comment by BertrandDekoninck** [comment]
• I miss the french mac keyboard in NextSpace Preferences app : it isn't listed, even if I selected it in Anaconda. Should I file an issue against Preferences ?

`Do you mean Apple keyboard with French layout? Then you should make setup in Keyboard Preferences: "Model" section for keyboard type and "Layouts" section for French layout.`

Yes, an Apple keyboard with French layout. It's listed as Mac variant in Anaconda layouts. I don't understand, which Keyboard Preferences you are talking about ? I meant NextSpace Preferences and I don't see a "Model" section in it. Only a layout section. See below :
![VirtualBox_NextSpace_05_03_2020_08_28_38](https://user-images.githubusercontent.com/12986802/75958089-ca0d8380-5ebb-11ea-8977-1f341fa977ef.png)

**Comment by trunkmaster** [comment]
@BertrandDekoninck I'm sorry. Everything I've taked about doesn't exist in 0.85 release. It's in master.

**Comment by nicktelindert** [comment]
BTW i take some time off, i probably start again next week. I probably am at week a lot the upcoming time, I expect my country to go in lockdown very soon. My government failed to respond in time and now the infections are rising very quickly. They expect that almost half of the country will be infected.

**Comment by alexmyczko** [comment]
Can you name your country?

**Comment by nicktelindert** [comment]
@alexmyczko the Netherlands, we are already out of testkits to test patients and medical personal.

**Comment by alexmyczko** [comment]
@nicktelindert Don't worry, we're not far behind: https://en.wikipedia.org/wiki/2020_coronavirus_pandemic_in_Europe#Pandemic_by_country (Switzerland)

**Comment by nicktelindert** [comment]
@alexmyczko Our real numbers are much higher than that, we have a lot of sick people which can not be confirmed because we can't test them. If we count the people who are suspected to have corona our numbers double. The amount of increase of sick people is now pretty much equal to italy. The governments plan is to slowly get the population infected in a controlled manner to build Herd immunity.

**Comment by alexmyczko** [comment]
that is pff topic but are you allowed to work from home? lockdown?

**Comment by nicktelindert** [comment]
@alexmyczko Yeah most people work from home now. To get this government plan to work we need at least 70% of the people to be infected. So i dont know how i feel about this plan.

**Comment by nicktelindert** [comment]
It just has been confirmed that we have 6000 infections, so much higher than expected. We started  out with a few hundred a few days ago.
